### PR TITLE
Configure test framework and make the tests pass

### PR DESCRIPTION
**Pull request by svilenvelikov**

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: ['jasmine', 'requirejs'],
+
+
+    // list of files / patterns to load in the browser
+    files: [
+      'test/test-main.js',
+      'src/js/lib/jquery/2.1.4/jquery.min.js',
+      'src/js/lib/angularjs/1.3.8/angular.js',
+      'src/js/lib/angularjs/1.3.8/angular-mocks.js',
```
Comment (mihailradkov) [inline karma.conf.js]: Maybe that could go in test/lib ?

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+
+    // start these browsers
+    // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
+    browsers: ['Chrome'],
+
+
+    // Continuous Integration mode
+    // if true, Karma captures browsers, runs the tests and exits
+    singleRun: false,
```
Comment (mihailradkov) [inline karma.conf.js]: Hm I would have to see how to override it for Travis CI

File: test/test-main.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+var TEST_REGEXP = /(spec|test)\.js$/i
+
+// console.log('karma files: ', window.__karma__.files);
+// Get a list of all the test files to include
+Object.keys(window.__karma__.files).forEach(function (file) {
+  if (TEST_REGEXP.test(file)) {
+    // Normalize paths to RequireJS module names.
+    // If you require sub-dependencies of test files to be loaded as-is (requiring file extension)
+    // then do not normalize the paths
+    // var normalizedTestModule = file.replace(/^\/base\/|\.js$/g, '')
```
Comment (mihailradkov) [inline test/test-main.js]: Remove commented code

File: karma.conf.js (hunk 1 above)
Comment (svilenvelikov) [reply karma.conf.js]: eventually, yes

File: test/test-main.js (hunk 3 above)
Comment (svilenvelikov) [reply test/test-main.js]: yeah

File: karma.conf.js (hunk 2 above)
Comment (mihailradkov) [reply karma.conf.js]: `"test": "karma start --single-run --browsers PhantomJS"` -> ~2 secs :D

**Comment by mihailradkov** [comment]
LGTM

**Pull request merged**